azurerm/data_source_netapp_pool.go

```diff
@@ -0,0 +1,88 @@
+package azurerm
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmNetAppPool() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmNetAppPoolRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[\da-zA-Z_-]{4,64}$`),
+					`The name must be between 4 and 64 characters in length and contain only letters, numbers, underscore or hyphens.`,
+				),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringMatch(
```

Comment:
could we make this a shared function in the netapp package and share it with the resource and netapp account resource?

Reply:
Moved this validation to netapp package as shared function.